modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1252 @@
+
+This new tree component is an important addition to the new Premium Plugin: *Advanced Templates*, which is now available as a premium plugin with editors running {productname} v6.4+.
+
+// https://notion.so/tinycloud/Four-words-to-live-by-96e307110aaa4137b94f9ba12874d480
+
+=== renderUI property in the Theme type can now return a Promise <RenderResult> instead of RenderResult
+
+// https://ephocks.atlassian.net/browse/TINY-9556
+In earlier versions of {productname}, the properties set for the `renderUI`, was causing issues with the theme rendering load times. As a result {productname} 6.4 added fixes to the `renderUI` properties to correct this issue.
+
+The changes saw the introduction of a new `renderUI` property called `Promise<RenderResult>` , which allows the current theme type to return a `Promise<RenderResult>` instead of the origional  `RenderResult`.
+
+In {productname} 6.4, the `renderUI` property in the Theme type, now returns a `Promise <RenderResult>`.
+// https://notion.so/tinycloud/Four-words-to-live-by-96e307110aaa4137b94f9ba12874d480
+
+=== New isEditable API to editor.selection that returns true or false if the current selection is editable
+
+// https://ephocks.atlassian.net/browse/TINY-9462
+// bforte this seems to be connected to https://ephocks.atlassian.net/browse/DOC-1888
+In previous versions of {productname}, elements that where within the `editor.selection` could be modified, if they sat within a HTML document body with the properties set to `contentEditable = false`.
+
+As a consequence, any `selection | node`, that was encapsulate within the editor body, could be affected by this such as:
+
+Expected output example:
+
+[source, HTML]
+----
+<p>Try to insert content here</p>
+----
+
+After executing a `onclick` event within the current `selection | node` using `<button onclick="tinymce.activeEditor.insertContent('text')">insertContent</button>`, the output would modify the existing `contentEditable = false;` element, which should not be possible.
+
+[source, HTML]
+----
+<p>Try to inserttextcontent here</p>
+----
+
+The new additions and improvements with the introduction of {productname} 6.4, include a new `isEditable` API been added to the `editor.selection` which now blocks the `editor.insertContent` API and `mceInsertContent` command when the contents return a `true` value if the editor detects the element is a `contentEditable = false;` element in a `contentEditable = false;` root.
+
+As a result, any `editor.selection` elements that return a `true` value when a specified element is `contentEditable = true;`, will call the new `isEditable` API to execute the new blocker.
```

Comment:
There is a lot of stuff here that is not relevant to demonstrating the bug. I want our release notes to be clear and concise. It looks like you've gotten hung up describing how the QA fiddle operates, not the actual details of the bug as covered in the JIRA docs field.
```suggestion
In previous versions of {productname}, when the body was set to `contentEditable = false` content could still be inserted with the `insertContent` API.

Example content:

[source, HTML]
----
<body contenteditable=false><p>Try to insert content here</p></body>
----

After selecting the space between "insert" and "content", executing `tinymce.activeEditor.insertContent('text')` would incorrectly insert the text which should not be possible.

[source, HTML]
----
<body contenteditable=false><p>Try to inserttextcontent here</p></body>
----

{productname} 6.4 includes a new `isEditable` method on the `editor.selection` API which is used to block the `editor.insertContent` API and `mceInsertContent` command when the selection covers a noneditable element.
```

Reply:
This suggestion has been added into the documentation.